js/base/Exchange.js

```diff
@@ -2361,36 +2361,33 @@ module.exports = class Exchange {
         return this.filterBySymbolSinceLimit (sorted, symbol, since, limit);
     }
 
-    isPostOnly (type, timeInForce = undefined, exchangeSpecificOption = undefined, params = {}) {
+    isPostOnly (type, timeInForce, params = {}) {
         /**
          * @ignore
          * @method
          * @param {string} type Order type
          * @param {string} timeInForce
-         * @param {boolean} exchangeSpecificOption True if the exchange specific post only setting is set
          * @param {dict} params Exchange specific params
          * @returns {boolean} true if a post only order, false otherwise
          */
         let postOnly = this.safeValue2 (params, 'postOnly', 'post_only', false);
-        params = this.omit (params, [ 'post_only', 'postOnly' ]);
-        const timeInForceUpper = (timeInForce !== undefined) ? timeInForce.toUpperCase () : undefined;
         const typeLower = type.toLowerCase ();
-        const ioc = timeInForceUpper === 'IOC';
-        const fok = timeInForceUpper === 'FOK';
-        const timeInForcePostOnly = timeInForceUpper === 'PO';
+        // we assume timeInForce is uppercase from safeStringUpper (params, 'timeInForce')
+        const ioc = timeInForce === 'IOC';
+        const fok = timeInForce === 'FOK';
+        const timeInForcePostOnly = timeInForce === 'PO';
         const isMarket = typeLower === 'market';
-        postOnly = postOnly || (typeLower === 'postonly') || timeInForcePostOnly || exchangeSpecificOption;
+        postOnly = postOnly || timeInForcePostOnly || (typeLower === 'postonly') || (typeLower === 'post_only');
         if (postOnly) {
             if (ioc || fok) {
                 throw new InvalidOrder (this.id + ' postOnly orders cannot have timeInForce equal to ' + timeInForce);
             } else if (isMarket) {
                 throw new InvalidOrder (this.id + ' postOnly orders cannot have type ' + type);
             } else {
-                timeInForce = timeInForcePostOnly ? undefined : timeInForce;
-                return [ 'limit', true, timeInForce, params ];
```

Comment:
if the user passed in `postOnly` as the type, then the type needs to be changed to limit, and if we omit this, it requires the developer using this method to remember to change the type, which is error prone